### ISSUE TITLE
Possible bug in meanteacher algorithm

### ISSUE THREAD

**Issue by 5v3D3** [microsoft/Semi-supervised-learning#102]
In meanteacher train_step (line 55), `logits_x_ulb_s` gets assigned the logits of `outs_x_ulb_w`. 
So the consistency loss is calculated only with `logits_x_ulb_w`, which results in 0 consistency loss in every iteration. 
`outs_x_ulb_s` is never used.

https://github.com/microsoft/Semi-supervised-learning/blob/9a24e00db040443b4b8d13ecb556b6948c56d15e/semilearn/algorithms/meanteacher/meanteacher.py#L54-L56

this should be:

```
outs_x_ulb_s = self.model(x_ulb_s) 
logits_x_ulb_s = outs_x_ulb_s['logits'] 
feats_x_ulb_s = outs_x_ulb_s['feat']
```

**Comment by Hhhhhhao** [comment]
This might be caused by a typo. Have you run the modified code? Does the result vary from the reported result?

**Comment by Hhhhhhao** [comment]
> Sorry but i haven't run the modified code on any of the benchmarks, since right now I am only working locally with my own dataset. With the modified code I get an `unsup_loss` that is greater than `0` though. So I can only suspect that it would perform better. But I had a look at your results and meanteacher performance looked a bit odd as it is quite similar to supervised (Both in classic_cv and usb_cv), while most other methods perform better . Which would make sense, if only the supervised loss is used in optimization.

Will check this in next update.

**Comment by Hhhhhhao** [comment]
Fixed in PR #135
